src/shell-app.c

```diff
@@ -1428,6 +1420,24 @@ shell_app_create_custom_launcher_with_name (ShellApp *app,
 
   new_path = g_build_filename (g_get_user_data_dir (), "applications", shell_app_get_id (app), NULL);
 
+  if (!g_file_test (new_path, G_FILE_TEST_EXISTS))
+    {
+      /* Create a new 'X-Endless-CreatedBy' key to indicate the file was created
+       * by eos-desktop */
+      g_key_file_set_string (keyfile,
```

Comment:
I think it should be fine to set this even when the file already exists. Otherwise if someone starts with a custom desktop file from a version that didn't have the marker, and then renamed the file again, they won't be able to remove it.